it/v1/endpoints/SubmitReturnControllerISpec.scala

```diff
@@ -0,0 +1,161 @@
+
+    val desResponse: JsValue = Json.parse(
+      s"""
+         |{
+         |  "processingDate": "2017-10-18T00:01:00Z",
+         |  "formBundleNumber": "123456789012",
+         |  "paymentIndicator": "DD",
+         |  "chargeRefNumber": "SKDJGFH9URGT"
+         |}
+         |""".stripMargin
+    )
+
+    val requestJson: JsValue = Json.parse(
+      s"""
+         |{
+         |        "periodKey" : "18A1",
+         |        "vatDueSales" : 1000,
+         |        "vatDueAcquisitions" : -1000,
+         |        "totalVatDue" : 0,
+         |        "vatReclaimedCurrPeriod" : 100,
+         |        "netVatDue" : 100,
+         |        "totalValueSalesExVAT" : 5000,
+         |        "totalValuePurchasesExVAT" : 1000,
+         |        "totalValueGoodsSuppliedExVAT" : 9999999999999,
+         |        "totalAcquisitionsExVAT" : 9999999999999,
+         |        "receivedAt":  "2020-05-05T12:01:00Z",
+         |        "uniqueId": "0123456789"
+         |}
+    """.stripMargin)
+
+    def uri: String = s"/$vrn/returns"
+    def desUrl: String = s"/enterprise/return/vat/$vrn"
+
+    def setupStubs(): StubMapping
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders((ACCEPT, "application/vnd.hmrc.1.0+json"))
+    }
+
+    def errorBody(code: String): String =
+    s"""
+       |      {
+       |        "code": "$code",
+       |        "reason": "des message"
+       |      }
+    """.stripMargin
+
+  }
+
+  "Submit Insolvent-Vat submitReturn endpoint" when {
+    "return a 201 status code with expected body" should {
+      "a valid request is made" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          DesStub.onSuccess(DesStub.POST, desUrl, OK, desResponse)
+        }
+
+        val response: WSResponse = await(request().post(requestJson))
+        response.status shouldBe CREATED
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    "return error according to spec" when {
+
+      def validationErrorTest(requestVrn: String,
+                              requestPeriodKey: String,
+                              expectedStatus: Int,
+                              expectedBody: MtdError): Unit = {
+        s"validation fails with ${expectedBody.code} error" in new Test {
+
+          override val vrn: String = requestVrn
+          override val periodKey: String = requestPeriodKey
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+          }
+
+          val response: WSResponse = await(request().post(requestJson))
+          response.status shouldBe expectedStatus
+          response.json shouldBe Json.toJson(expectedBody)
+          response.header("Content-Type") shouldBe Some("application/json")
+        }
+      }
+
+      val input = Seq(
+        ("badVrn", "AA11", BAD_REQUEST, VrnFormatError)
+      )
```

Comment:
All the validation errors should be tested here, it's missing a test for 

FORMAT_PERIOD_KEY
FORMAT_VALUE
FORMAT_UNIQUE_ID
FORMAT_RECEIVED_AT